xpsparser/parser.py

```diff
@@ -22,42 +22,580 @@
+                for key, value in obj.__dict__.items() 
+                if not callable(value) and not key.startswith('_')])
+            if classkey is not None and hasattr(obj, "__class__"):
+                data[classkey] = obj.__class__.__name__
+            return data
+        else:
+            return obj
+        
+    def writeJSON(self, filename):
+        """ Write the currently loaded data-set to a JSON file."""
+        with open(f'{filename}.json', 'w') as fp:
+            json.dump(self.dataset, fp, indent=4)
+            
+    def _moveChannelMetaToGlobal(self, dataset):
+        """ Move the channel metadata into the spectrum's global metadata.
+        
+        This is a bandaid fix to re-organize the data structure.
+        """
+        for d in dataset:
+            d['metadata']['data_labels'] = []
+            d['metadata']['device_settings'] = []
+            channel_values = []
+            for s in d['data']:
+                channel_id = s['channel_id']
+                s['device_settings']['channel_id'] = channel_id
+                d['metadata']['device_settings']+=[s['device_settings']]
+                d['metadata']['data_labels'] += [{'channel_id':channel_id,
+                                                  'label':s['label'],
+                                                  'unit':s['unit']}]
+                channel_values += [s['values']]
+            d['data'] = channel_values
+        return dataset
+    
+    def removeAlign(self):
+        """ Remove the 'align' spectra from the data-set."""
+        for idx, d in reversed(list(enumerate(self.dataset))):
+            if 'align' in d['metadata']['spectrum_region'].lower():
+                del self.dataset[idx]
+                
+    def extractAllTags(self):
+        """ Extract the user-entered metadata tags."""
+        for spectrum in self.dataset:
+            self._extractTags(spectrum)
+    
+    def _extractTags(self, spectrum):
+        tags = spectrum['metadata']['group_name'].split('#')
+        spectrum['metadata']['experiment_parameters'] = {}
+        vals = ''
+        if len(tags)>1:
+            for t in tags[1:]:
+                key_val = t.split(':')
+                key = key_val[0].strip()
+                
+                val = key_val[1].strip().strip(',')
+    
+                if len(val) != 0:
+                    spectrum['metadata']['experiment_parameters'][key] = val
+                    vals += val + ', '
+            if len(tags[0].strip()) == 0:
+                group_name = vals
+            else:
+                group_name = tags[0]
+            spectrum['metadata']['group_name'] = group_name
+            
+    def convertToJson(self, filepath):
+        """ Convert directly from XY to JSON."""
+        self.parseFile(filepath)
+        self.removeAlign()
+        self.extractAllTags()
+        name = filepath.rsplit('.',1)[0]
+        self.writeJSON(name)
+
+
+
+class XPSParser(MatchingParser):
```

Comment:
Both classes should be joined into a single class and methods `ProdigyParserXY.parseFile` and `XPSParser.run` should be joined as well. Make sure to clear all members before run to make the parser re-usable.